cmd/sovereignnode/chainSimulator/tests/bridge/common.go

```diff
@@ -61,3 +69,70 @@ func DeploySovereignBridgeSetup(
 		FeeMarketAddress: feeMarketAddress,
 	}
 }
+
+// Deposit will deposit tokens in the bridge sc safe contract
+func Deposit(
+	t *testing.T,
+	cs chainSim.ChainSimulator,
+	sender []byte,
+	nonce *uint64,
+	contract []byte,
+	tokens []chainSim.ArgsDepositToken,
+	receiver []byte,
+	transferData *sovereign.TransferData,
+) {
+	require.True(t, len(tokens) > 0)
+
+	dtaCodec := cs.GetNodeHandler(core.SovereignChainShardId).GetRunTypeComponents().DataCodecHandler()
+
+	depositArgs := core.BuiltInFunctionMultiESDTNFTTransfer +
+		"@" + hex.EncodeToString(contract) +
+		"@" + fmt.Sprintf("%02X", len(tokens))
+
+	for _, token := range tokens {
+		depositArgs = depositArgs +
+			"@" + hex.EncodeToString([]byte(token.Identifier)) +
+			"@" + getNonceArg(t, dtaCodec, token.Nonce) +
+			"@" + hex.EncodeToString(token.Amount.Bytes())
+	}
+
+	transferDataArg := getTransferDataArg(t, dtaCodec, transferData)
+	depositArgs = depositArgs +
+		"@" + hex.EncodeToString([]byte(deposit)) +
+		"@" + hex.EncodeToString(receiver) +
+		transferDataArg
```

Comment:
Generally speaking, it's possible to do this using the serializer, as well. Can also stay as it is.

Reply:
This sounds good. I will refactor this.